c2rust-transpile/src/diagnostics.rs

```diff
@@ -42,7 +42,7 @@ pub fn init(mut enabled_warnings: HashSet<Diagnostic>, log_level: log::LevelFilt
                 Level::Trace => "trace",
             };
             let target = record.target();
-            let warn_flag = if let Ok(_) = Diagnostic::from_str(target) {
+            let warn_flag = if Diagnostic::from_str(target).is_ok() {
                 format!(" [-W{}]", target)
             } else {
                 String::new()
```

Comment:
Could be
```rust
let warn_flag = Diagnostic::from_str(target).map(|_| format!(" [-W{}]", target)).unwrap_or_default();
```

Reply:
Fixed by https://github.com/immunant/c2rust/pull/460/commits/3ac46569f1955c3e06b6c770ebc1c04568dafffa.

Reply:
I will disagree here. Using the `Result` combinators in a context where you don't actually use any of the payloads and just treat the `Result` as a boolean is confusing. The first version makes it clear that the contents of the parse are irrelevant, and isn't significantly worse looking.